joint_state_publisher/joint_state_publisher/joint_state_publisher.py

```diff
@@ -90,59 +90,66 @@ def init_urdf(self, robot):
         for child in robot.childNodes:
             if child.nodeType is child.TEXT_NODE:
                 continue
-            if child.localName == 'joint':
-                jtype = child.getAttribute('type')
-                if jtype in ['fixed', 'floating', 'planar']:
-                    continue
-                name = child.getAttribute('name')
-                self.joint_list.append(name)
-                if jtype == 'continuous':
-                    minval = -math.pi
-                    maxval = math.pi
-                else:
-                    try:
-                        limit = child.getElementsByTagName('limit')[0]
+            if child.localName != 'joint':
+                continue
+            jtype = child.getAttribute('type')
+            if jtype in ['fixed', 'floating', 'planar']:
+                continue
+            name = child.getAttribute('name')
+            if jtype == 'continuous':
+                minval = -math.pi
+                maxval = math.pi
+            else:
+                try:
+                    limit_list = child.getElementsByTagName('limit')
+                    if limit_list:
+                        limit = limit_list[0]
```

Comment:
```suggestion
                    limit_list = child.getElementsByTagName('limit')[0]
```

(this will retain the previous semantics of skipping the joint if there are no limits)